package/MDAnalysis/auxiliary/EDR.py

```diff
@@ -0,0 +1,305 @@
+        try:
+            return self._data[key]
+        except KeyError:
+            raise KeyError(f"'{key}' is not a key in the data_dict dictionary."
+                           " Check the EDRReader.terms attribute")
+
+
+class EDRReader(base.AuxReader):
+    """ Auxiliary reader to read data from an .edr file.
+
+    Default reader for .edr files. All data from the file will be read and
+    stored on initialisation.
+
+    Parameters
+    ----------
+    filename : str
+        Location of the file containing the auxiliary data.
+    **kwargs
+       Other AuxReader options.
+
+    See Also
+    --------
+    :class:`~MDAnalysis.auxiliary.base.AuxReader`
+    :func:`~MDAnalysis.coordinates.base.add_auxiliary`
+
+    Note
+    ----
+    The file is assumed to be of a size such that reading and storing the full
+    contents is practical.
+    """
+
+    format = "EDR"
+    _Auxstep = EDRStep
+
+    def __init__(self, filename: str, **kwargs):
+        self._auxdata = Path(filename).resolve()
+        self.data_dict = pyedr.edr_to_dict(filename)
+        self._n_steps = len(self.data_dict["Time"])
+        # attribute to communicate found energy terms to user
+        self.terms = list(self.data_dict.keys())
+        super(EDRReader, self).__init__(**kwargs)
+
+    def _read_next_step(self) -> EDRStep:
+        """Read next auxiliary step and update ``auxstep``.
+
+        Returns
+        -------
+        AuxStep object
+            Updated with the data for the new step.
+
+        Raises
+        ------
+        StopIteration
+            When end of auxiliary data set is reached.
+        """
+        auxstep = self.auxstep
+        new_step = self.step + 1
+        if new_step < self.n_steps:
+            auxstep._data = {term: self.data_dict[term][self.step + 1]
+                             for term in self.terms}
+            auxstep.step = new_step
+            return auxstep
+        else:
+            self.rewind()
+            raise StopIteration
+
+    def _go_to_step(self, i: int) -> EDRStep:
+        """ Move to and read i-th auxiliary step.
+
+        Parameters
+        ----------
+        i: int
+            Step number (0-indexed) to move to
+
+        Returns
+        -------
+        :class:`EDRStep`
+
+        Raises
+        ------
+        ValueError
+            If step index not in valid range.
+        """
+        if i >= self.n_steps or i < 0:
+            raise ValueError("Step index {0} is not valid for auxiliary "
+                             "(num. steps {1})".format(i, self.n_steps))
+        self.auxstep.step = i - 1
+        self.next()
+        return self.auxstep
+
+    def read_all_times(self) -> np.ndarray:
+        """ Get list of time at each step.
+
+        Returns
+        -------
+        NumPy array of float
+            Time at each step.
+        """
+        return self.data_dict[self.time_selector]
+
+    def get_data(self, data_selector: Union[str, List[str], None] = None) \
```

Comment:
also `data_selector` is a bit clunky imho (you nearly expect a method to be passed). Are we baked into having this name for the argument or could we have `data_keys` or some "key" variant (which better represents the fact that these are keys to `data_dict`)?

Reply:
`data_selector` is named here to match the rest of `auxiliary`. Both `data_selector` and `time_selector` are introduced in the base class and documented like this: 
```python
    @property
    def data_selector(self):
        """Key(s) to select auxiliary data values of interest from the full set
        of data read for each step (as stored in ``auxstep``).

        Type differs between auxiliary formats, depending how the data for each
        step is read in and stored - e.g. data from .xvg files is read in as
        a list and `data_selector` must be a list of valid indicies. If data
        selection is not enabled by the reader, ``data_selector`` will default
        to ``None``.

        See each individual auxiliary reader.
        """
```

Calling it keys makes sense for EDRReaders as we are dealing with dictionaries, but it is less appropriate for the XVGReader and the future NumPy reader. So I would prefer not to change the name across all of auxiliary, but it could be done for EDR. Should I prioritise matching base API here, or giving the clearer name?